logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/FeatureConfigEventReceiver.kt

```diff
@@ -19,219 +19,77 @@
-                )
-            }
-
-            is Event.FeatureConfig.MLSUpdated -> {
-                val mlsEnabled = event.model.status == Status.ENABLED
-                val selfUserIsWhitelisted = event.model.allowedUsers.contains(selfUserId.toPlainID())
-                userConfigRepository.setMLSEnabled(mlsEnabled && selfUserIsWhitelisted)
-
-                kaliumLogger.logEventProcessing(
-                    EventLoggingStatus.SUCCESS,
-                    event
-                )
-            }
-
-            is Event.FeatureConfig.ClassifiedDomainsUpdated -> {
-                val classifiedDomainsEnabled = event.model.status == Status.ENABLED
-                userConfigRepository.setClassifiedDomainsStatus(classifiedDomainsEnabled, event.model.config.domains)
-
-                kaliumLogger.logEventProcessing(
-                    EventLoggingStatus.SUCCESS,
-                    event
-                )
-            }
-
-            is Event.FeatureConfig.ConferenceCallingUpdated -> {
-                val conferenceCallingEnabled = event.model.status == Status.ENABLED
-                userConfigRepository.setConferenceCallingEnabled(conferenceCallingEnabled)
-
-                kaliumLogger.logEventProcessing(
-                    EventLoggingStatus.SUCCESS,
-                    event
-                )
-            }
-
-            is Event.FeatureConfig.GuestRoomLinkUpdated -> {
-                handleGuestRoomLinkFeatureConfig(event.model.status)
-
+            .onSuccess {
                 kaliumLogger.logEventProcessing(
                     EventLoggingStatus.SUCCESS,
                     event
                 )
             }
-
-            is Event.FeatureConfig.SelfDeletingMessagesConfig -> {
-                handleSelfDeletingFeatureConfig(event.model)
-
-                kaliumLogger.logEventProcessing(
-                    EventLoggingStatus.SUCCESS,
-                    event,
-                    Pair("isDurationEnforced", (event.model.config.enforcedTimeoutSeconds ?: 0) > 0),
-                )
-            }
-
-            is Event.FeatureConfig.MLSE2EIUpdated -> {
-                val gracePeriodEndMs = event.model.config
-                    .verificationExpirationNS
-                    .toDuration(DurationUnit.NANOSECONDS)
-                    .inWholeMilliseconds
-
-                userConfigRepository.setE2EISettings(
-                    E2EISettings(
-                        isRequired = event.model.status == Status.ENABLED,
-                        discoverUrl = event.model.config.discoverUrl,
-                        gracePeriodEnd = Instant.fromEpochMilliseconds(gracePeriodEndMs)
+            .onFailure {
+                if (it is CoreFailure.FeatureNotImplemented) {
+                    kaliumLogger.logEventProcessing(
+                        EventLoggingStatus.SKIPPED,
+                        event,
+                        Pair("info", "Ignoring unknown feature config update")
```

Comment:
Why don't we move these logs to each individual handler? I can't see any specific log referencing to each receiver class so we should be good to move it?

Reply:
What do I gain by moving this to each handler? Don't I need to repeat these logs for each handler then?

Reply:
well, as a starter not having to add the same log to 2 different receivers, FeatureConfig and SyncFeatureConfig. Then if for any reason, we decide to start using these handlers in different places as well, then we will not need to add the logs on the invoking class.

Reply:
OK, I see your point. We can refactor the again if it becomes and issue. I'll keep it like it is for now since the logging for events and for slow sync are different.